### ISSUE TITLE
Progress bar issue

### ISSUE THREAD

**Issue by reza-sohrabi** [keras-team/keras#10876]
I opened another issue long ago regarding this, but seems like it is not prioritized, but I think it should be. The problem is with some custom loss functions and their calculation in the progress bar. 
It is my understanding that the progress bar calculates the current loss of the epoch by averaging out the loss of all the batches that have been used. For example, if the `mse` of batch 1 is 2.1 and the `mse` of batch 2 is 1.9, progress bar shows `(2.1+1.9)/2 = 2` as the current `mse` of the epoch. This is fine for mse as the last operation in mse is `K.mean()` which is linear. But this approach results in the wrong calculations when the last operation is something nonlinear like `K.sqrt()` like in `rmse` .
Let me give you a simple example to see the problem. Suppose the batch size is 2 and `y=[1,2]` and `y_pred = [1.2, 1.8] ` for batch 1 and `y=[0.7,2]`, `y_pred = [1.1, 2] ` for batch 2. What is the `rmse` of the batches and the overall `rmse` of the epoch at each batch:
batch 1: `rmse_batch= sqrt((0.2^2+0.2^2)/2) = sqrt(0.04), rmse_epoch = sqrt(0.04)`  
batch 2: `rmse_batch = sqrt((0.4^2+0^2)/2) = sqrt(0.08), rmse_epoch = sqrt((0.2^2+0.2^2+0.4^2+0^2)/4) = sqrt(0.06)`  
While this is shown in the progress bar:
batch 1: ` rmse_epoch =sqrt(0.04)`  which is correct
batch 2: ` rmse_epoch =(sqrt(0.04)+sqrt(0.08))/2 != sqrt(0.06)`  which is not correct.
I suggest for the loss functions with nonlinear functions at the end, Keras should do the calculation of the epoch loss differently. It adds some overhead but not that much.

**Comment by brge17** [comment]
The progress bar shows batchwise averages, which is intuitive.

The model has changed weights  between ``batch_1`` and ``batch_2``, so what you are proposing is not valid.

**Comment by reza-sohrabi-ucr** [comment]
Yes, you are right, my approach was wrong, but anyways, Keras does not give an intuitive result for `rmse`.
